Java/Concentus/src/main/java/org/concentus/OpusDecoder.java

```diff
@@ -708,6 +705,39 @@ public int decode(byte[] in_data, int in_data_offset,
         }
     }
 
+    /**
+     * Decodes an Opus packet.
+     * @param in_data The input payload. This may be NULL if that previous packet was lost in transit (when PLC is enabled)
+     * @param in_data_offset The offset to use when reading the input payload. Usually 0
+     * @param len The number of bytes in the payload (the packet size)
+     * @param out_pcm A buffer to put the output PCM, in a byte array. The output size is (# of samples) * (# of channels) * 2.
+     *      You can use the OpusPacketInfo helpers to get a hint of the frame size before you decode the packet if you need exact sizing.
+     * @param out_pcm_offset The offset to use when writing to the output buffer
+     * @param frame_size The number of samples (per channel) of available space in the output PCM buf.
+     * If this is less than the maximum packet duration (120ms; 5760 for 48khz), this function will
+     * not be capable of decoding some packets. In the case of PLC (data == NULL) or FEC (decode_fec == true),
+     * then frame_size needs to be exactly the duration of the audio that is missing, otherwise the decoder will
+     * not be in an optimal state to decode the next incoming packet. For the PLC and FEC cases, frame_size *must*
+     * be a multiple of 2.5 ms.
+     * @param decode_fec Indicates that we want to recreate the PREVIOUS (lost) packet using FEC data from THIS packet. Using this packet
+     * recovery scheme, you will actually decode this packet twice, first with decode_fec TRUE and then again with FALSE. If FEC data is not
+     * available in this packet, the decoder will simply generate a best-effort recreation of the lost packet. In that case,
+     * the length of frame_size must be EXACTLY the length of the audio that was lost, or else the decoder will be in an inconsistent state.
+     * @return The number of decoded samples (per channel)
+     * @throws OpusException 
+     */
+    public int decode(byte[] in_data, int in_data_offset,
+            int len, byte[] out_pcm, int out_pcm_offset, int frame_size, boolean decode_fec) throws OpusException {
+    	short[] spcm = new short[out_pcm.length / 2];
+    	int decSamples = decode(in_data, in_data_offset, len, spcm, 0, frame_size, decode_fec);
+    	//Convert short array to byte array
+    	for (int c = out_pcm_offset; c < spcm.length; c++) {

```

Comment:
No, this won't work either because the offset needs to be applied to the out_pcm index but not spcm, so you can't just modify the range of c. It would actually be a lot easier (and faster) if you just created a separate index.

```
int pcm_idx = out_pcm_offset;
for (int c = 0; c < spcm.length; c++) {
    out_pcm[pcm_idx++] = (byte) (spcm[c] & 0xff);
    out_pcm[pcm_idx++] = (byte) ((spcm[c] >> 8) & 0xff);
```